apps/concierge_site/lib/plugs/rate_limit.ex

```diff
@@ -0,0 +1,43 @@
+defmodule ConciergeSite.Plugs.RateLimit do
+  @moduledoc "Limits IPs or users to a reasonable number of requests per unit of time."
+
+  import Plug.Conn
+  @behaviour Plug
+
+  @limit_anonymous 5
+  @limit_signed_in 15
+  @one_minute 60_000
+
+  @doc """
+  Initialize the plug. If the option `enable?: false` is given, no actual rate limiting will be
+  performed except for requests with a `rate_limit?: true` assign in the conn.
```

Comment:
**suggestion (non-blocking):** Personally I would find it helpful to have a super quick hint on why/when the `rate_limit?: true` condition might occur. Is it just for testing purposes?